### DISCUSSION TITLE
can  any db model convert to pydantic

### DISCUSSION THREAD

**Discussion by LLjiahai** [tiangolo/pydantic-sqlalchemy#96] [Questions] [answered]
import peewee as pw
class Order(pw.Model):
    project_id = pw.CharField(null=True, max_length=255, verbose_name="")
    order_id = pw.CharField(null=True, max_length=255, index=True, verbose_name="")

pydanticA = sqlalchemy_to_pydantic(Order)

**Comment by ljluestc** [comment]
import peewee as pw
from pydantic import BaseModel

class Order(pw.Model):
    project_id = pw.CharField(null=True, max_length=255, verbose_name="")
    order_id = pw.CharField(null=True, max_length=255, index=True, verbose_name="")

class OrderPydantic(BaseModel):
    project_id: str
    order_id: str

order_instance = Order(project_id="123", order_id="456")
pydantic_instance = OrderPydantic(project_id=order_instance.project_id, order_id=order_instance.order_id)

print(pydantic_instance)

**Comment by tiangolo** [comment, marked as answer]
Nope, this is for SQLAlchemy, not for Peewee.

---

Thanks for the patience with my reply! :sweat_smile:

Just so you know, I deprecated this library a while ago in favor of SQLModel which would solve all the same use cases in a much better way, and I'm directing all the effort in that direction.

SQLModel uses Pydantic models to define both data and table (SQL) models, so it's easier to do all these things. :rocket:
